test/unit/bulk_write.test.js

```diff
@@ -131,4 +135,48 @@ describe('Bulk Writes', function() {
 
     expect(() => result.insertedIds).to.not.throw();
   });
+
+  describe('#mergeBatchResults', function() {
+    context('when opTime is an object', function() {
+      context('when the lastOp is a Timestamp', function() {
+        const batch = [];
+        const bulkResult = {
+          ok: 1,
+          writeErrors: [],
+          writeConcernErrors: [],
+          insertedIds: [],
+          nInserted: 0,
+          nUpserted: 0,
+          nMatched: 0,
+          nModified: 0,
+          nRemoved: 1,
+          upserted: [],
+          lastOp: {
+            ts: 7020546605669417496,
+            t: 10
+          }
+        };
+        const result = {
+          n: 8,
+          nModified: 8,
+          opTime: Timestamp.fromNumber(8020546605669417496),
+          electionId: '7fffffff0000000000000028',
+          ok: 1,
+          $clusterTime: {
+            clusterTime: '7020546605669417498',
+            signature: {
+              hash: 'AAAAAAAAAAAAAAAAAAAAAAAAAAA=',
+              keyId: 0
+            }
+          },
+          operationTime: '7020546605669417498'
+        };
+
+        it('replaces the lastOp with the properly formatted timestamp', function() {
+          mergeBatchResults(batch, bulkResult, null, result);
+          expect(bulkResult.lastOp.t).to.equal(Long.ZERO);
```

Comment:
Can we add few more assertions here to catch us. lastOp will now always be transformed to have the shape of {`t` and `ts`} right?

```suggestion
          expect(bulkResult.lastOp).to.be.an('object');
          expect(bulkResult.lastOp).to.have.property('t', Long.ZERO);
	  expect(bulkResult.lastOp).to.have.property('ts');

```

Reply:
Can we not also assert on the `ts` property?